### PR TITLE
more information about library dependencies

### PR DESCRIPTION
**Pull request by laerus**
in request of #1118 on why would someone want to install the extra dependencies

### CODE REVIEW

**Comment by adqm** [comment]
Thanks for putting this in, @laerus.  Rather than just adding a description of what each of the optional packages is, can we also add a brief description of what, specifically, will change within xonsh if that is enabled (e.g., prompt_toolkit allows syntax highlighting of input and more interactive tab completion, etc).

**Comment by laerus** [comment]
should i add it inline, inside the parenthesis, or as a new pargraph?

File: docs/dependencies.rst [hunk 1]
```diff
@@ -5,11 +5,11 @@ Xonsh currently has the following external dependencies,
 *Run Time:*
 
     #. Python v3.4+
-    #. PLY (optional, included with xonsh)
-    #. prompt-toolkit (optional)
-    #. Jupyter (optional)
-    #. setproctitle (optional)
-    #. distro (optional)
+    #. PLY (optional, included with xonsh, alternative parsing tool)
```
Comment (gforsyth) [inline docs/dependencies.rst]: PLY isn't an alternative parsing tool, it's just that we vend it ourselves in case it isn't installed locally.

Comment (laerus) [inline docs/dependencies.rst]: so should i remote the optional part aswell?

**Comment by gforsyth** [comment]
This looks good, @laerus -- additionally, would you mind adding a "Note" under `prompt_toolkit` that states that if you use `prompt_toolkit` then `pygments` is also a dependency?  I feel like we've been bitten by that one a few too many times now

**Comment by gforsyth** [comment]
Awesome, thanks @laerus ! This is good to go once Travis finishes up.

**Pull request merged**